WordPress/gutenberg
# Add an alt attribute to the linked featured image if there is no post title

## What?

In [#41077](https://github.com/WordPress/gutenberg/issues/41077) it was reported that when a post or page has no title,
but has a linked featured image, there is no link text. This issue only affects the front, not the editors.

This PR adds an alt attribute to the post featured image if it is linked and the post title is missing.
The new alt text is "Untitled post" followed by the post id.

## Why?
The missing link text is an accessibility issue since there is no context for the link.

## Testing Instructions
1. Create a new post but do not add a title.
2. Add a post featured image block and enable the "Link to post" setting in the block sidebar. Save.
3. View the single post on the front and confirm that the featured image has the correct alt text.
4. View a page with a query loop with featured images. Confirm that the featured image for your post has the correct alt text.

### Testing Instructions for screen reader users
After creating a new post as described above, go to the front view of the post.
Remember that the post has no title, and depending on the theme template, the single post may display 
a featured image above the post content. The image that you want to test is the linked image and it should be the only item in the post content.
Confirm that both the link and the image are announced, followed by "Untitled post number".

Voiceover announces "Visited, link, image, Untitled post 720".

## Screenshots or screencast
